src/metatensor/models/utils/data/readers/readers.py

```diff
@@ -0,0 +1,223 @@
+    """
+    return _base_reader(
+        readers=STRESS_READERS,
+        filename=filename,
+        fileformat=fileformat,
+        key=target_value,
+    )
+
+
+def read_structures(
+    filename: str,
+    fileformat: Optional[str] = None,
+) -> List[System]:
+    """Read structure informations from a file.
+
+    :param filename: name of the file to read
+    :param fileformat: format of the structure file. If :py:obj:`None` the format is
+        determined from the suffix.
+    :returns: list of structures
+    """
+    return _base_reader(
+        readers=STRUCTURE_READERS, filename=filename, fileformat=fileformat
+    )
+
+
+def read_virial(
+    filename: str,
+    target_value: str,
+    fileformat: Optional[str] = None,
+) -> TensorBlock:
+    """Read virial informations from a file.
+
+    :param filename: name of the file to read
+    :param target_value: target value key name to be parsed from the file.
+    :param fileformat: format of the structure file. If :py:obj:`None` the format is
+        determined from the suffix.
+    :returns: target value stored stored as a :class:`metatensor.TensorBlock`
+    """
+    return _base_reader(
+        readers=VIRIAL_READERS,
+        filename=filename,
+        fileformat=fileformat,
+        key=target_value,
+    )
+
+
+def read_targets(conf: DictConfig) -> Dict[str, TensorMap]:
+    """Reading all target information from a fully expanded config.
+
+    :param conf: config containing the keys for what should be read.
+    :returns: Dictionary containing one TensorMaps for each target section in the
+        config."""
+    target_dictionary = {}
+
+    for target_key, target in conf.items():
+        if target["quantity"] == "energy":
+            block = read_energy(
+                filename=target["read_from"],
+                target_value=target["key"],
+                fileformat=target["file_format"],
+            )
+
+            if target["forces"]:
+                try:
+                    position_gradient = read_forces(
+                        filename=target["forces"]["read_from"],
+                        target_value=target["forces"]["key"],
+                        fileformat=target["forces"]["file_format"],
+                    )
+                except KeyError:
+                    logger.warning(
+                        f"Forces not found in section {target_key!r}. "
+                        "Continue without forces!"
+                    )
+                else:
+                    logger.info(
+                        f"Forces found in section {target_key!r}. Forces are taken for "
+                        "training!"
+                    )
+                    block.add_gradient(
+                        parameter="positions", gradient=position_gradient
+                    )
+
+            if target["stress"] and target["virial"]:
+                raise ValueError(
+                    "Cannot add gradient displacement gradient for stress and virial!"
```

Comment:
I think most people wouldn't understand this. How about "cannot use stress and virial at the same time" and/or "only one of the two is allowed at once"?

Reply:
Agree. I will change it.

Reply:
FYI: I think I'll add an explicit gradient w.r.t. `"strain"` (i.e. `-stress`) in rascaline, to resolve an ambiguity in the `"cell"` gradients. A similar naming could be used internally here! (no comment on the user facing error message)